### PR TITLE
Destination Snowflake: SqlGenerator class cleanup

### PR DESCRIPTION
**Pull request by gisripa**
## What
Cleaning up `SnowflakeSqlGenerator.kt` by replacing `apache.commons.text.StringSubstitutor` with Kotlin's native String interpolation.  There is a lot of duplication of code snippets but did not touch it in this PR to minimize diffs. Cannot remove the `apache.commons.text` dependency in gradle yet since it is still used in `SnowflakeDestinationHandler.kt`

## Review guide
No functional or query changes. Any diffs should be solely nuking `StringSubtitutor` references and replacing with Kotlin's string interpolation and simplifying String concatenation from collections into Kotlin idioms

## User Impact
None. Zero. Zilch. 

## Can this PR be safely reverted and rolled back?

- [x] YES 💚
- [ ] NO ❌

### CODE REVIEW

**Comment by gisripa** [comment]
* **#43348** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/43348?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @gisripa and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: airbyte-integrations/connectors/destination-snowflake/src/main/kotlin/io/airbyte/integrations/destination/snowflake/typing_deduping/SnowflakeSqlGenerator.kt [hunk 1]
```diff
@@ -297,189 +232,122 @@ class SnowflakeSqlGenerator(private val retentionPeriodDays: Int) : SqlGenerator
...
-                            """
-                CASE
-                  WHEN (TYPEOF("_airbyte_data":"${'$'}{raw_col_name}") NOT IN ('NULL', 'NULL_VALUE'))
-                    AND (${'$'}{json_extract} IS NULL)
-                    THEN OBJECT_CONSTRUCT('field', '${'$'}{printable_col_name}', 'change', 'NULLED', 'reason', 'DESTINATION_TYPECAST_ERROR')
-                  ELSE NULL
-                END
-                """.trimIndent()
-                        )
+            stream.columns.entries.joinToString(
```
Comment (stephane-airbyte) [inline airbyte-integrations/connectors/destination-snowflake/src/main/kotlin/io/airbyte/integrations/destination/snowflake/typing_deduping/SnowflakeSqlGenerator.kt:239]: nit: use the compact form

File: airbyte-integrations/connectors/destination-snowflake/src/main/kotlin/io/airbyte/integrations/destination/snowflake/typing_deduping/SnowflakeSqlGenerator.kt [hunk 2]
```diff
@@ -490,42 +358,27 @@ class SnowflakeSqlGenerator(private val retentionPeriodDays: Int) : SqlGenerator
...
-                ) as row_number FROM ${'$'}{final_table_id}
-              )
-              WHERE row_number != 1
-            );
-            
-            """.trimIndent()
-            )
+        return """
+        |DELETE FROM 
+        |  ${id.finalTableId(QUOTE, finalSuffix.uppercase(Locale.getDefault()))}
```
Comment (stephane-airbyte) [inline airbyte-integrations/connectors/destination-snowflake/src/main/kotlin/io/airbyte/integrations/destination/snowflake/typing_deduping/SnowflakeSqlGenerator.kt:368]: when the same expression is several times in the same string (or across several strings), I'd rather see it in its own variable. I think it also makes the string template easier to read

Comment (stephane-airbyte) [reply airbyte-integrations/connectors/destination-snowflake/src/main/kotlin/io/airbyte/integrations/destination/snowflake/typing_deduping/SnowflakeSqlGenerator.kt:368]: we could also add a function that does the `id.finalTableId.etc1

**Review by stephane-airbyte — APPROVED** [review]
minor comments. No need to re-submit for review

File: airbyte-integrations/connectors/destination-snowflake/src/main/kotlin/io/airbyte/integrations/destination/snowflake/typing_deduping/SnowflakeSqlGenerator.kt [hunk 3]
```diff
@@ -82,48 +79,28 @@ class SnowflakeSqlGenerator(private val retentionPeriodDays: Int) : SqlGenerator
     }
 
     override fun createSchema(schema: String): Sql {
-        return of(
-            StringSubstitutor(java.util.Map.of("schema", StringUtils.wrap(schema, QUOTE)))
-                .replace("CREATE SCHEMA IF NOT EXISTS \${schema};")
-        )
+        return of("CREATE SCHEMA IF NOT EXISTS \"$schema\";")
```
Comment (edgao) [inline airbyte-integrations/connectors/destination-snowflake/src/main/kotlin/io/airbyte/integrations/destination/snowflake/typing_deduping/SnowflakeSqlGenerator.kt:82]: nit: triplequoted strings could avoid the escape
```suggestion
        return of("""CREATE SCHEMA IF NOT EXISTS "$schema";""")
```

File: airbyte-integrations/connectors/destination-snowflake/src/main/kotlin/io/airbyte/integrations/destination/snowflake/typing_deduping/SnowflakeSqlGenerator.kt (hunk 2 above)
Comment (gisripa) [reply airbyte-integrations/connectors/destination-snowflake/src/main/kotlin/io/airbyte/integrations/destination/snowflake/typing_deduping/SnowflakeSqlGenerator.kt:368]: Yup. there are many such duplicated snippets. Tried avoiding them but these little ones probably won't hurt much.

**Review by edgao — APPROVED** [review]
I only skimmed this but it looks generally right + the sqlgenerator tests should protect us. one nit about avoiding `\"`, otherwise lgtm

File: airbyte-integrations/connectors/destination-snowflake/src/main/kotlin/io/airbyte/integrations/destination/snowflake/typing_deduping/SnowflakeSqlGenerator.kt (hunk 1 above)
Comment (gisripa) [reply airbyte-integrations/connectors/destination-snowflake/src/main/kotlin/io/airbyte/integrations/destination/snowflake/typing_deduping/SnowflakeSqlGenerator.kt]: done.

File: airbyte-integrations/connectors/destination-snowflake/src/main/kotlin/io/airbyte/integrations/destination/snowflake/typing_deduping/SnowflakeSqlGenerator.kt (hunk 2 above)
Comment (gisripa) [reply airbyte-integrations/connectors/destination-snowflake/src/main/kotlin/io/airbyte/integrations/destination/snowflake/typing_deduping/SnowflakeSqlGenerator.kt]: fixed this particular one. I'll do more cleanup when introducing `MERGE`

**Pull request merged**